app/gui/src/project-view/components/GraphEditor.vue

```diff
@@ -402,18 +402,36 @@ const codeEditorHandler = codeEditorBindings.handler({
 
 // === Documentation Editor ===
 
-const displayedDocs = ref<SuggestionId>()
+const overrideDisplayedDocs = ref<SuggestionId>()
 const aiMode = ref<boolean>(false)
+const docsForSelection = computed(() => {
+  const selected = nodeSelection.tryGetSoleSelection()
+  if (!selected.ok) return Err('Select a single component to display help')
+  const suggestionId = graphStore.db.nodeMainSuggestionId.lookup(selected.value)
+  if (suggestionId == null) return Err('No documentation available for selected component')
+  return Ok(suggestionId)
+})
+const displayedDocs = computed(() =>
+  overrideDisplayedDocs.value ? Ok(overrideDisplayedDocs.value) : docsForSelection.value,
+)
+
+watchEffect(() => {
+  const projectId = projectStore.id as ProjectId
```

Comment:
This cast should not be necessary. Instead, the store should already hold a value of `ProjectId` type.

Reply:
I expect it to be quite a thorough refactor.